core/primitives/src/state_patch.rs

```diff
@@ -0,0 +1,26 @@
+use crate::state_record::StateRecord;
+
+/// Changes to the state to be applied via sandbox-only state patching
+/// functionality.
+///
+/// As we only expose this functionality for sandbox, we make sure that the
+/// object is only constructable if sandbox feature is enabled.
+pub struct StatePatch {
+    records: Vec<StateRecord>,
+}
+
+impl StatePatch {
+    // NB: it's crucial that all creation APIs are guarded with `#[cfg(feature = "sandbox")]`.
```

Comment:
'records' is a public field, so people could create StatePatch as:

StatePatch {records: XX} 

right?